### PR TITLE
[WOR-265] Add comment explaining why invalid region test remains ignored.

### PR DESCRIPTION
**Pull request by cahrens**
This is now a very boring PR, but hopefully it will help folks in the future.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

[WOR-265]: https://broadworkbench.atlassian.net/browse/WOR-265?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by cahrens** [comment]
jenkins retest

**Comment by cahrens** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala [hunk 1]
```diff
@@ -201,8 +201,7 @@ class WorkspaceApiSpec
         }(owner.makeAuthToken(billingScopes))
 
         exception.fields("statusCode").convertTo[Int] should equal(400)
-        exception.fields("message").convertTo[String] should startWith("Workspace creation failed. Error trying to create bucket ")
-        exception.fields("message").convertTo[String] should endWith regex(s" in Google project (.+) in region `${invalidRegion}`.".r)
+        exception.fields("message").convertTo[String] should startWith("Workspace bucket location must be a single region of format")
```
Comment (cahrens) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala]: Note that we already have test coverage [here](https://github.com/broadinstitute/rawls/blob/30555ad9af1b48c68cd6f32c3130a4e3dbd24d11/core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala#L992) for a create workspace request with an invalid region and [here](https://github.com/broadinstitute/rawls/blob/30555ad9af1b48c68cd6f32c3130a4e3dbd24d11/core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/WorkspaceApiServiceSpec.scala#L1435) for clone workspace request. Is there value in also keeping this test, which does the creation through a call to  `Orchestration.workspaces.create` vs. sending a `WorkspaceRequest` to the workspaces API endpoints?

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala [hunk 2]
```diff
@@ -191,6 +191,13 @@ class WorkspaceApiSpec
...
       "to get an error message when they try to create a workspace with a bucket region that is invalid" ignore {
         implicit val token: AuthToken = ownerAuthToken
+        // Note that this invalid region passes the regexp in `withWorkspaceBucketRegionCheck`, so workspace creation is
+        // attempted and fails with the bucket creation error. However, due to bug WOR-296, `withTemporaryBillingProject`
+        // is unable to delete the temporary project, causing the test to fail (when this test was first introduced,
+        // there was a different way of creating test workspaces that did not involve creating and deleting test billing
+        // projects). There is already test coverage of `withWorkspaceBucketRegionCheck` in `WorkspaceApiServiceSpec`,
+        // so there is no point in changing the invalid region name to one detected by the regexp, as that would change
+        // the code path executed (no workspace creation would be attempted) and simply duplicate existing unittests.
         val invalidRegion = "invalid-region1"
```
Comment (cahrens) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala:201]: [WOR-296](https://broadworkbench.atlassian.net/browse/WOR-296) is the bug to make sure that orphaned resources are not left behind if workspace creation fails

**Review by marctalbott — APPROVED** [review]

**Review by aherbst-broad — APPROVED** [review]

**Pull request merged**